src/components/App/index.tsx

```diff
@@ -150,57 +148,43 @@ export default class App extends Component<Props, State> {
     }
   }
 
-  onChange(
-    index: 0 | 1,
-    preprocessorState: PreprocessorState,
-    type: EncoderType,
-    options?: EncoderOptions,
-  ): void {
-    // Some type cheating here.
-    // encoderMap[type].defaultOptions is always safe.
-    // options should always be correct for the type, but TypeScript isn't smart enough.
-    const encoderState: EncoderState = {
-      type,
-      options: options || encoderMap[type].defaultOptions,
-    } as EncoderState;
-
-    const images = cleanMerge(this.state.images, index, { encoderState, preprocessorState });
-    this.setState({ images });
-  }
```

Comment:
It seems better to update the relevant parts of state in the individual change events.